_posts/2018-11-06-simple-slackbot.md

```diff
@@ -0,0 +1,55 @@
+---
+layout: post
+title:  "Simple Slackbot"
+date: 2018-11-06 00:03:30 +0200
+categories: javascript
+author_name: Sergii Paryzhskyi
+author_url : /author/sergii_paryzhskyi
+author_avatar: sergii_paryzhskyi
+read_time : 7
+show_related_posts: false
+square_related: recommend-wolf
+feature_image: posts/2018-11-06-simple-slackbot/poster.jpg
+---
+
+Recently I've built a small slackbot that is posting with certain regularity in specified slack-channel. I will tell you about the idea behind it and also share some technical details.
+
+
+## The Idea
+
+The idea to build such a bot came from one colleague of mine - [Jacek Smolak][jacek]. At that point he was reading the book ["97 Things Every Programmer Should Know"][book-link]. In order to share those programmer tips with others and also remind them to himself, he came up with idea of posting each of them one by one on a daily basis in appropriate slack channel. For us this slack channel is #craft, you probably have something similar amoung your channels in slack.
+
+
+## Implementation
+
+The implementation is dead simple. As it was written with JavaScript, the easiest way was to use an official [slack npm package][slack-npm]. Posting something in specific channel becoming a one-liner that returns a Promise that we can handle depending if it is success or failure:
+
+```js
+slackClient.chat.postMessage({
+    channel: '#craft',
+    message: '...'
+});
+```
+
+The very same interface is used for sending direct messages, so without changing anything you can specify your own slack-name instead of channel. In this case you will be getting the messages from the bot directly. Its quite handy for testing purposes as well.
```

Comment:
What is quite handy? The fact that it is the same interface or getting the message from the bot directly? Or something else?
What I mean by that is that the last sentence is unclear.
Also you jump to testing all of a sudden without any deeper dive (if needed). I would either skip that sentence or extend it.

Reply:
What I've tried to express is that you can build something to post to a channel, but if you want to test or try it out, you can actually change the name of the channel to your own name in Slack and you can get then messages directly

Reply:
And the fact that its possible without rewriting code, but by changing ENV, its quite handy

Reply:
I would explain it then, as I was thinking about (code) unit testing, but that can be just me being tests junkie ;)

Reply:
yep, I've just added a clarification that its about manual testing